From: Gerald Pfeifer <gerald@pfeifer.com>
Subject: xcopy: Add an explicit check to avoid an out of array access. access.
Message-Id: <E1UJlQt-0007W1-7w@wine.codeweavers.com>
Date: Sun, 24 Mar 2013 01:52:23 +0100 (CET)

This _may_ not be necessary, strictly speaking, but new versions of
GCC (GCC 4.8) complain about this. (And there might be a fringe case?)

Gerald

---
 programs/xcopy/xcopy.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/programs/xcopy/xcopy.c b/programs/xcopy/xcopy.c
index 7a68831..83ef973 100644
--- a/programs/xcopy/xcopy.c
+++ b/programs/xcopy/xcopy.c
@@ -272,7 +272,8 @@ static BOOL XCOPY_ProcessExcludeFile(WCHAR* filename, WCHAR* endOfName) {
         int length = lstrlenW(buffer);
 
         /* Strip CRLF */
-        buffer[length-1] = 0x00;
+        if( length > 0 )
+          buffer[length-1] = 0x00;
 
         /* If more than CRLF */
         if (length > 1) {
-- 
1.8.1.3
